sbncode/CAFMaker/CAFMaker_module.cc

```diff
@@ -774,9 +775,17 @@ void CAFMaker::produce(art::Event& evt) noexcept {
   }
 
   // Prep truth-to-reco-matching info
-  std::map<int, std::vector<std::pair<geo::WireID, const sim::IDE*>>> id_to_ide_map = PrepSimChannels(simchannels, *geometry);
-  std::map<int, std::vector<art::Ptr<recob::Hit>>> id_to_truehit_map = PrepTrueHits(hits, clock_data, *bt_serv.get());
-  std::map<int, caf::HitsEnergy> id_to_hit_energy_map = SetupIDHitEnergyMap(hits, clock_data, *bt_serv.get());
+  std::map<int, std::vector<std::pair<geo::WireID, const sim::IDE*>>> id_to_ide_map;
+  std::map<int, std::vector<art::Ptr<recob::Hit>>> id_to_truehit_map;
+  std::map<int, caf::HitsEnergy> id_to_hit_energy_map;
+
+  if ( !fIsRealData ) {
+    art::ServiceHandle<cheat::BackTrackerService> bt_serv;
+
+    id_to_ide_map = PrepSimChannels(simchannels, *geometry);
+    id_to_truehit_map = PrepTrueHits(hits, clock_data, *bt_serv.get());
```

Comment:
The `.get()` should not be necessary here:
```suggestion
    id_to_truehit_map = PrepTrueHits(hits, clock_data, *bt_serv);
```
(`ServiceHandle::operator *` will return a reference to the service object directly).
I _suggest_ removal.

Reply:
Sure - I just copied the previous function call into the loop, but I can make this change. Are the other instances of `.get()` similar? (I thought I spotted a few last night...)

Reply:
Indeed there were multiple and multiple `*pi_serv.get()`. But since they are all service handle `.get()` calls with the `*` operator, I assume they can all be changed in this fashion. If this is not the case let me know!